File: ServerApp/src/main/java/lk/cooperative/cooperativejaela/entity/Storereturn.java
package lk.cooperative.cooperativejaela.entity;

import javax.persistence.*;
import java.sql.Date;
import java.util.Collection;
import java.util.Objects;

@Entity
public class Storereturn {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name = "id")
    private int id;
    @Basic
    @Column(name = "date")
    private Date date;
    @Basic
    @Column(name = "description")
    private String description;
    @OneToMany(mappedBy = "storereturn")
    private Collection<Sritem> sritems;
    @ManyToOne
    @JoinColumn(name = "employee_id", referencedColumnName = "id", nullable = false)
    private Employee employee;
    @ManyToOne
    @JoinColumn(name = "store_id", referencedColumnName = "id", nullable = false)
    private Store store;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Storereturn that = (Storereturn) o;
        return id == that.id && Objects.equals(date, that.date) && Objects.equals(description, that.description);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, date, description);
    }

    public Collection<Sritem> getSritems() {
        return sritems;
    }

    public void setSritems(Collection<Sritem> sritems) {
        this.sritems = sritems;
    }

    public Employee getEmployee() {
        return employee;
    }

    public void setEmployee(Employee employee) {
        this.employee = employee;
    }

    public Store getStore() {
        return store;
    }

    public void setStore(Store store) {
        this.store = store;
    }
}

File: ClientApp/src/app/entity/storereturn.ts
import { Employee } from "./employee";
import { Shopstatus } from "./shopstatus";
import {Route} from "./route";
import {Disorder} from "./disorder";
import {Store} from "./store";
import {Sritem} from "./sritem";
import {Disorderitem} from "./disorderitem";

export class Storereturn{

  public id !: number;
  public date !: string;
  public description !: string;
  public employee !: Employee;
  public store !: Store;
  public sritems !: Array<Sritem >;

  constructor(id: number, date: string, description: string, employee: Employee, store: Store, sritems: Array<Sritem>) {
    this.id = id;
    this.date = date;
    this.description = description;
    this.employee = employee;
    this.store = store;
    this.sritems = sritems;
  }

}





